klippy/extras/display/st7920.py

```diff
@@ -135,13 +135,14 @@ def write_graphics(self, x, y, row, data):
             gfx_fb -= 32
             x += 16
         self.graphics_framebuffers[gfx_fb][x:x+len(data)] = data
+    def set_glyphs(self, glyphs):
+        for glyph_name, glyph_data in glyphs.items():
+            self.icons[glyph_name] = glyph_data
+        # Setup animated glyphs
+        self.cache_glyph('fan2', 'fan1', 0)
+        self.cache_glyph('bed_heat2', 'bed_heat1', 1)
     def write_glyph(self, x, y, glyph_name):
-        glyph_id = self.cached_glyphs.get(glyph_name)
-        if glyph_id is not None and x & 1 == 0:
-            # Render cached icon using character generator
-            glyph_name = glyph_id[0]
-            self.write_text(x, y, glyph_id[1])
```

Comment:
FYI, this should also not be deleted in order to have the caching optimization.

-Kevin

Reply:
sorry, abotu that. I'll fix it right away.